Update Windows CI workflow for PHP 8.4

cmb69 opened the pull request:
We also update to the latest actions:

* setup-php-sdk v9 is required for PHP 8.4 support
* actions/checkout v4 and actions/upload-artifact v4 are not deprecated

---

Not sure whether the uploaded artifacts still make sense. That might have been added since until recently DLLs have not been shown on https://pecl.php.net/package/parallel. On the other hand, these artifacts may be helpful for Windows users testing a bug fix or new feature.

cmb69 committed:
Update Windows CI workflow for PHP 8.4

We also update to the latest actions:

* setup-php-sdk v9 is required for PHP 8.4 support
* actions/checkout v4 and actions/upload-artifact v4 are not deprecated

realFlowControl commented:
> Not sure whether the uploaded artifacts still make sense. That might have been added since until recently DLLs have not been shown on https://pecl.php.net/package/parallel. On the other hand, these artifacts may be helpful for Windows users testing a bug fix or new feature.

I do not understand this comment, sorry. Can you elaborate?

cmb69 commented:
See https://github.com/krakjoe/parallel/actions/runs/10253935939?pr=318. There are downloads of the Windows DLLs (the same as the downloads via PECL). These are only useful for testing purposes nowadays, since https://github.com/php/web-pecl/pull/97 had been merged (previously, the parallel DLLs have not been displayed on the PECL package page).

cmb69 commented:
Oh, I've just noticed that Windows CI still uses pthreadsV2; should probably be updated to pthreadsV3. Need to check that out.

cmb69 committed:
Update to pthreads V3

We also use the new download server to fetch that dependency.

cmb69 committed:
Use windows-2022 instead of windows-latest

That might prevent issues when a new windows runner is made available
by Github.

cmb69 commented:
> Oh, I've just noticed that Windows CI still uses pthreadsV2; should probably be updated to pthreadsV3. Need to check that out.

Indeed, the PECL DLLs (at least as of PHP 8.0) are also using V3, so I've updated that, and also switched to windows-2022, to prevent potential problems with a new runner.

realFlowControl commented:
Thank you ❤️
